database-core/database-task.py

```diff
@@ -0,0 +1,170 @@
+from typing import List, Type
+from sqlalchemy import create_engine, Column, String, Integer, update, delete
+from sqlalchemy.orm import registry
+from sqlalchemy.orm import sessionmaker
+import argparse
+import sys
+
+DATABASE_NAME = 'films_db.db'
+TABLE_NAME = 'films'
+
+mapper_registry = registry()
+Base = mapper_registry.generate_base()
+engine = create_engine(f'sqlite+pysqlite:///{DATABASE_NAME}', echo=True)
+Session = sessionmaker(bind=engine)
+
+
+class Film (Base):
+    __tablename__ = TABLE_NAME
+
+    id = Column('id', Integer, primary_key=True, nullable=False)
+    title = Column('title', String)
+    director = Column('director', String)
+    release_year = Column('release_year', Integer)
+
+    def __init__(self, film_id, title, director, release_year):
+        self.id = film_id
+        self.title = title
+        self.director = director
+        self.release_year = release_year
+
+    def __repr__(self):
+        return f'{self.id} {self.title} {self.director} {self.release_year}'
+
+
+def input_films(qty: int) -> List[Film]:
+    """
+    Gets user input for 'qty' iterations and returns list of Film instances.
+    """
+    films = []
+    for n in range(qty):
+        print(f'Please enter film data({n+1})')
+        film = Film(
+            film_id=int(input('id: ')),
+            title=input('title: '),
+            director=input('director: '),
+            release_year=int(input('release_year: '))
+        )
+        films.append(film)
+    return films
+
+
+def create_table() -> None:
+    """
+    Creates table using global engine and meta_data
+    """
+    Base.metadata.create_all(bind=engine)
+
+
+def add_films(*films: Film) -> None:
+    """
+    Adds films to the 'films' table
+    """
+
+    session = Session()
+    for film in films:
+        session.add(film)
+
+    session.commit()
+
+
+def get_all_films() -> List[Type[Film]]:
+    """
+    Returns all existing films
+    """
+    session = Session()
+    return session.query(Film).all()
```

Comment:
As in the previous method, it is worth taking care to close the session either by `session.close()` , or by using the context manager `with` to close it automatically